exchange/xgrid.F90

```diff
@@ -1593,9 +1595,18 @@ subroutine setup_xmap(xmap, grid_ids, grid_domains, grid_file, atm_grid, lnd_ug_
   endif
 
   call mpp_clock_begin(id_load_xgrid)
-  do g=1,size(grid_ids(:))
-     grid => xmap%grids(g)
-     if (g==1) grid1 => xmap%grids(g)
+
+  ! nvhpc compiler workaround
+  ! saves passed in xmap as an allocatable
+  ! this lets us use a normal assignment instead of a pointer assignment in the loop to avoid a compiler error
```

Comment:
Do we need to report this to nvidia?

Reply:
Let's make sure we have tested using the very latest nvhpc 24.3 before filing a bug with NVidia.

Reply:
The xgrid compilation error was pretty hard to reproduce, Niki first saw this using 22.3 and I've only been able to get the error using his xml to compile. I don't seem to get the error when i compile with autotools, and didn't even get it when i tried compiling one of my xml's with nvidia.